index.bs

```diff
@@ -1118,6 +1118,60 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 :: A [=credential property=] is some characteristic property of a [=public key credential source=], such as whether it is a
     [=client-side discoverable credential=] or a [=server-side credential=].
 
+: <dfn>Credential Record</dfn>
+:: In order to implement the algorithms defined in [[#sctn-rp-operations]],
+    the [=[RP]=] MUST store some properties of registered [=public key credential sources=].
+    The [=credential record=] [=struct=] is an abstraction of these properties stored in a [=user account=].
+    A&nbsp;credential record is created during a [=registration ceremony=] and used in subsequent [=authentication ceremonies=].
+    [=[RPS]=] MAY delete credential records as necessary or when requested by users.
+
+    The following [=struct/items=] are RECOMMENDED in order to implement all steps of
+    [[#sctn-registering-a-new-credential]] and [[#sctn-verifying-assertion]] as defined:
+
+    <dl dfn-for="credential record">
+        :   <dfn>type</dfn>
+        ::  The [=public key credential source/type=] of the [=public key credential source=].
+
+        :   <dfn>id</dfn>
+        ::  The [=Credential ID=] of the [=public key credential source=].
+
+        :   <dfn>publicKey</dfn>
+        ::  The [=credential public key=] of the [=public key credential source=].
+
+        :   <dfn>signCount</dfn>
+        ::  The latest value of the [=authData/signCount|signature counter=] in the [=authenticator data=]
+            from any [=ceremony=] using the [=public key credential source=].
+
+        :   <dfn>transports</dfn>
+        ::  The value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            when the [=public key credential source=] was [=registration|registered=].
+
+            Note: Modifying or removing [=list/items=] from the value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            could negatively impact user experience, or even prevent use of the corresponding credential.
+
+        :   <dfn>BE</dfn>
+        ::  The value of the [=authData/flags/BE=] [=flag=] when the [=public key credential source=] was created.
+
+        :   <dfn>BS</dfn>
+        ::  The latest value of the [=authData/flags/BS=] [=flag=] in the [=authenticator data=]
+            from any [=ceremony=] using the [=public key credential source=].
+    </dl>
+
+    [=[RPS]=] MAY include any additional [=struct/items=] as needed,
```

Comment:
Should we recommend that if the device public key is requested and returned, it should be stored with or with a reference to the credential record?

Reply:
Hm, good point. I'm not sure it's appropriate to call out specific extensions here, but we should at least point out that extensions might add items that need to be stored. Perhaps as an extension point similar to how extensions add fields to `partial dictionary AuthenticationExtensionsClientOutputs` etc?

Reply:
we probably also need to store the attestation ca chain or related attestation metadata to assist with DPK here. We already store this in webauthn-rs, here: https://github.com/kanidm/webauthn-rs/blob/master/webauthn-rs-core/src/interface.rs#L245

Reply:
(Irrespective of DPK, some enterprise RPs certainly like to store the attestation for record keeping.)

Reply:
Should be fixed by a5710f6, please re-review!